From fb83d623958b67c5335c18233eacee4cb577d164 Mon Sep 17 00:00:00 2001
From: Beniamino Galvani <bgalvani@redhat.com>
Date: Fri, 3 Aug 2018 10:54:17 +0200
Subject: [PATCH 12/14] device: clear queued IP config sources when the device
 is unrealized

If the device is later realized again, we assert that there aren't any
IP config changes queued. Therefore, they must be cleared on
unrealize().

(cherry picked from commit 9ed07fbb46c7dfbd4db8f1a908d4a522dc4917b4)
---
 src/devices/nm-device.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/devices/nm-device.c b/src/devices/nm-device.c
index 19e43762b..52ed22717 100644
--- a/src/devices/nm-device.c
+++ b/src/devices/nm-device.c
@@ -4236,6 +4236,9 @@ nm_device_unrealize (NMDevice *self, gboolean remove_resources, GError **error)
 		}
 	}
 
+	nm_clear_g_source (&priv->queued_ip_config_id_4);
+	nm_clear_g_source (&priv->queued_ip_config_id_6);
+
 	g_object_freeze_notify (G_OBJECT (self));
 	NM_DEVICE_GET_CLASS (self)->unrealize_notify (self);
 
-- 
2.17.0

